### PR TITLE
Issue/6719 featured image picker

### LINKED ISSUE #6719: Featured Image: launch the new media picker
For setting a Featured Image in a Post, the app currently shows the Media Browser for the user to pick an existing picture in the blog's WP Media Library.

It'd be great to let the user choose their Featured Image from any other sources, and if such a choice is not in their Media Library, make the process seamless and upload the image for them.

1. user opens / starts a post / draft
2. user taps on the settings icon
3. user scrolls down to the “featured image” section and taps to add / change
4. HERE instead of launching the media browser (where you can only pick existing pics from the Media Library), we launch the current Media Picker used in the Editor. This way we’re offering the user all the possibilities they have.
5. In case the user chooses to use a picture that is not already in their Media Library (i.e. a new photo, for instance), we take care of that seamlessly in the background through async upload.

### PR DESCRIPTION
**Pull request by nbradbury**
Fixes #6719 - previously you could only select a featured image from the WP media library, this PR changes this to show the new device media picker when choosing a featured image (and from there you can view your WP media library if desired).

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/util/WPMediaUtils.java [hunk 1]
```diff
@@ -441,4 +443,62 @@ public static boolean currentUserCanDeleteMedia(@NonNull SiteModel site) {
     public static int getFlingDistanceToDisableThumbLoading(@NonNull Context context) {
         return ViewConfiguration.get(context).getScaledMaximumFlingVelocity() / 2;
     }
+
+    /*
+     * returns a MediaModel from a device media URI
+     */
+    public static MediaModel mediaModelFromLocalUri(@NonNull Context context,
```
Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/util/WPMediaUtils.java]: I think this method belongs to / would be a great addition to `FluxCUtils.java` in the FluxC library. Can we move it there? We can open a separate issue as well

Comment (nbradbury) [reply WordPress/src/main/java/org/wordpress/android/util/WPMediaUtils.java]: Sure, we can move that to FluxC. I'd prefer that as a separate issue, though.

Comment (mzorz) [reply WordPress/src/main/java/org/wordpress/android/util/WPMediaUtils.java]: 👍

**Comment by mzorz** [comment]
Found this:

1. start draft
2. tap on settings icon
3. tap “set featured image”
4. the media picker appears
5. choose an image
6. the “Uploading media…” dialog appears
7. rotate the device
8. the dialog is gone and you’re presented with the media picker, where you were
9. nothing else happens, tapping BACK gets you to the settings screen but no featured image is set.

Checking the logs I can see both in step 7 or 9 the upload still is happening. If you don’t tap BACK on step 9 but rather stay there until you see the upload is complete in the Android logs, then you’ll see everything completes successfully and you’re automatically taken to the Settings screen with the newly uploaded image set as the featured image.

Maybe it’d be good to try and handle the dialog state on device rotation, so we can show it at all times, preventing interaction.

**Idea for another PR:**
It would be nice to let the UploadService handle everything in the background and save the featured image when it’s done uploading, without blocking interaction

I imagine this:
1. in the settings screen, tap on “Set featured image”
2. media picker appears, and lets the user choose
3. immediately after choosing, the user is taken back to the Post settings screen and they can see the image in the  “featured image” section, with an “Uploading” overlay, just like what we do in the Media Browser screen
4. once the upload is finished, the image is presented there without overlays, just like what is shown there today.

We can fix the rotation/dialog problem in this PR and merge, and leave this other one for another PR, what do you think @nbradbury ?

**Comment by nbradbury** [comment]
> We can fix the rotation/dialog problem in this PR and merge, and leave this other one for another PR, what do you think @nbradbury ?

Agreed, and I'm embarrassed I didn't think to rotate while that dialog was showing - noob mistake.

**Comment by nbradbury** [comment]
@mzorz I fixed the problem with the disappearing progress dialog in 311f98b

**Comment by mzorz** [comment]
LGTM :shipit:

**Pull request merged**